packages/vite/src/node/plugins/dynamicImportVars.ts

```diff
@@ -198,11 +200,22 @@ export function dynamicImportVarsPlugin(config: ResolvedConfig): Plugin {
 
         const { rawPattern, glob } = result
 
+        let newPattern = posix.relative(
+          posix.dirname(importer),
+          await toAbsoluteGlob(rawPattern, config.root, importer, (im) =>
+            this.resolve(im, importer).then((i) => i?.id || im)
+          )
+        )
+
+        if (!/^\.{1,2}\//.test(newPattern)) {
+          newPattern = `./${newPattern}`
+        }
+
```

Comment:
Can you add the `snapslot` of the output(packages/vite/src/node/__tests__/plugins/dynamicImportVar)?

Reply:
Of course!